yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -573,10 +573,19 @@ object ApplicationMaster extends Logging {
   }
 
   private[spark] def sparkContextInitialized(sc: SparkContext): Unit = {
+    if(master == null) {
```

Comment:
(Should be a space after `if` but that can be fixed on merge; might fix that if you make any other change.) 

I think the text is OK although I suppose that as of 1.4 there's the launcher API, but, probably not worth writing a man page here.